app/scripts/modules/core/src/pipeline/filter/ExecutionFilters.tsx

```diff
@@ -99,6 +106,49 @@ export class ExecutionFilters extends React.Component<IExecutionFiltersProps, IE
       ExecutionFilterService.updateExecutionGroups(this.props.application);
       this.props.setReloadingForFilters(false);
     });
+    this.getFalconFilters();
+  };
+  private getFalconFilters = (): any => {
+    const { application } = this.props;
+    const sortFilter = ExecutionState.filterModel.asFilterModel.sortFilter;
+    if (application.pipelineConfigs.loadFailure) {
+      return [];
+    }
+    const configs = get(application, `pipelineConfigs.data`, []);
+    // let filterOn = configs.map(config => config.filterOn).filter(x => x);
+    const app = configs.map((config) => config.filterOn).filter((x) => x);
+    if (!app.length) return null;
+
+    const customFilters = ExecutionFilterService.getRelations();
+
+    const FILTER_PARAMS: any = {};
+    const Filters: any = {};
+
+    flatten(customFilters).forEach((x: any) => {
+      FILTER_PARAMS[x] = Object.keys(sortFilter[x]);
+      Filters[x] = {};
+    });
+
+    customFilters.forEach((filters: string[]) => {
+      let filterOn = app;
+      filters.forEach((x, index) => {
+        if (index > 0) {
```

Comment:
customFilters is an array of array of string, eg: [["Type"],["FI","FD","Cell"],["SI"]].
The elements in filters are in a hierarchical relationship, e.g FI -> FD -> Cell. Rest are individual e.g Types and SI.
We set filterOn for all but the first index because we need to show only those options that are still available in the selected filter. e.g if I select an FI then only those options are showed in the FD and Cell that are present in the current filtered Executions

Reply:
Got it. I think it is worth adding a comment to explain that either here in the interface for the custom filters.